pkg/reconciler/v1alpha1/taskrun/taskrun.go

```diff
@@ -349,19 +349,22 @@ func updateStatusFromPod(taskRun *v1alpha1.TaskRun, pod *corev1.Pod) {
 
 	switch pod.Status.Phase {
 	case corev1.PodRunning:
+		taskRun.Status.Phase = "Running"
```

Comment:
Instead of adding a new field could we leverage `reason` field from `Condition`? :thinking:

Reply:
@hrishin 
Thank you very much for the comments.
For the `Condition` in the `Status` section, I still not really get the point, I guess it's something like `Pod Condition`, supply some information about pass/not pass an internal state/step of entity(pod or taskrun).  If that's true, the `Reason` from `Condition` is just an internal stuff, should not present a "high-level summary", just like what "Phase" do.

Reply:
Quite a valid point, thank you :+1: :slightly_smiling_face:  

@vdemeester just wanondering what is the orginal intent of the `Reason` field in the `Condition` struct? :thinking: 
I'm unable to find the refences for the `Reason` in the code. Its getting [updated](https://github.com/tektoncd/pipeline/blob/master/pkg/reconciler/v1alpha1/taskrun/taskrun.go#L355) in some places.